File: services/scala-play/services/check/CheckService.scala
package services.check

import com.google.inject.ImplementedBy
import scala.concurrent.Future
import models._

@ImplementedBy(classOf[CheckService])
trait ICheckService {
  def checkEmpty(): Future[CheckEmptyResponse]
  def checkForbidden(): Future[CheckForbiddenResponse]
  def sameOperationName(): Future[SameOperationNameResponse]
}

sealed trait CheckEmptyResponse
object CheckEmptyResponse {
  case class Ok() extends CheckEmptyResponse
}

sealed trait CheckForbiddenResponse
object CheckForbiddenResponse {
  case class Ok(body: Message) extends CheckForbiddenResponse
  case class Forbidden() extends CheckForbiddenResponse
}

sealed trait SameOperationNameResponse
object SameOperationNameResponse {
  case class Ok() extends SameOperationNameResponse
  case class Forbidden() extends SameOperationNameResponse
}

File: services/ts-koa-superstruct/developed/check_service.ts
import * as service from './../generated/check'
import * as models from './../generated/models'

export const checkService = (): service.CheckService => {
    const checkEmpty = async (): Promise<void> => {
        throw new Error('Not Implemented')
    }

    const checkEmptyResponse = async (params: service.CheckEmptyResponseParams): Promise<void> => {
        throw new Error('Not Implemented')
    }

    const checkForbidden = async (): Promise<service.CheckForbiddenResponse> => {
        throw new Error('Not Implemented')
    }

    const sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        throw new Error('Not Implemented')
    }

    return {checkEmpty, checkEmptyResponse, checkForbidden, sameOperationName}
}
